pkg/operator/ceph/cluster/mgr/mgr.go

```diff
@@ -305,16 +307,29 @@ func (c *Cluster) configureMgrModules() error {
 			return errors.New("name not specified for the mgr module configuration")
 		}
 		if wellKnownModule(module.Name) {
-			return errors.Errorf("cannot configure mgr module %s that is configured with other cluster settings", module.Name)
+			return errors.Errorf("cannot configure mgr module %q that is configured with other cluster settings", module.Name)
 		}
 		minVersion, versionOK := c.moduleMeetsMinVersion(module.Name)
 		if !versionOK {
-			return errors.Errorf("module %s cannot be configured because it requires at least Ceph version %+v", module.Name, minVersion)
+			return errors.Errorf("module %q cannot be configured because it requires at least Ceph version %q", module.Name, minVersion.String())
 		}
 
 		if module.Enabled {
+			if module.Name == balancerModuleName {
+				// Set mon compat client to luminous before enabling the balancer mode "upmap"
+				err := client.SetMinCompatClientLuminous(c.context, c.Namespace)
```

Comment:
Why do we need to set the min client to luminous? What's the default min client?

Reply:
In my minikube cluster I see a jewel client. Any idea how to tell where that client is coming from? Old kernel version?
```
[root@rook-ceph-tools-78b599b6dd-lrzkb /]# ceph features
{
...
 "client": [
        {
            "features": "0x7010fb86aa42ada",
            "release": "jewel",
            "num": 1
        },
        {
            "features": "0x3ffddff8ffacffff",
            "release": "luminous",
            "num": 2
        }
```

Reply:
@jdurgin

Reply:
Yes old client Kernel.

Reply:
We need to enable the min client to luminous because that's a prerequisite for the upmap balancer to work, that's it I guess. I don't have all the details.

Reply:
Ok let's go ahead with the PR. Since the balancer won't be enabled by default yet, the users will have to opt-in to this behavior, so everyone wouldn't be impacted by the luminous client requirement.

Reply:
That's a kernel client - there's was an issue with how it reported itself (showing up as jewel) even though does support upmap (and has since 4.13 upstream, and centos/rhel 7.5). So it will work fine with the upmap balancer.

Ilya describes this on ceph-users here: https://lists.ceph.io/hyperkitty/list/ceph-users@ceph.io/message/JP44CGR4FO5Y7FMCC3VYDRXNQ3T26UHY/